### PR TITLE
[WIP] Firearms-131: Add logic to retry keycloak auth attempts in case of an…

### PR DESCRIPTION
**Pull request by Alex-Swann**
… auth fail when setting up bearer tokens for uploaded files

### CODE REVIEW

File: apps/common/models/file-upload.js [hunk 1]
```diff
@@ -29,14 +29,40 @@ module.exports = class UploadModel extends Model {
...
-      return this.auth().then(bearer => {
-        data.url = (data.url.replace('/file', '/vault')) + '&token=' + bearer.bearer;
+    }).then(async data => {
+      try {
+        const token = await this.authWithRetries(config.keycloak.authTokenRetries);
+        data.url = (data.url.replace('/file', '/vault')) + `&token=${token}`;
         return data;
-      });
+      } catch (e) {
+        throw e
```
Comment (sulthan-ahmed) [inline apps/common/models/file-upload.js:38]: Is it worth adding a descriptive error message here?

Comment (Alex-Swann) [reply apps/common/models/file-upload.js:38]: see line 57

File: apps/common/models/file-upload.js [hunk 2]
```diff
@@ -29,14 +29,40 @@ module.exports = class UploadModel extends Model {
...
+          break;
+        }
+
+        if (retries <= 0) {
+          throw new Error('Failed to authenticate the upload. Please try again.');
+        }
+      }
+      return bearer.bearer;
+    } catch (e) {
+      throw e
```
[the remark this answers is not in the document]
Comment (Alex-Swann) [reply apps/common/models/file-upload.js:62]: see line 57

File: apps/common/models/file-upload.js (hunk 1 above)
Comment (sulthan-ahmed) [reply apps/common/models/file-upload.js:38]: o ok, never seen that before.  I guess if it catches all cases, that's all good 👍

**Review by sulthan-ahmed — APPROVED** [review]
lgtm

**Comment by sulthan-ahmed** [comment]
I'll leave you to look at the failing CI

**Comment by Alex-Swann** [comment]
@sulthan-ahmed yeah I need to update unit tests then will get @MCrawleyHomeOffice to give it a once over

**Pull request closed without merging**